plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -409,11 +409,21 @@ func (c *CiscoTelemetryMDT) handleTelemetry(data []byte) {
 		if content != nil {
 			// Parse values
 			for _, subfield := range content.Fields {
+				if len(subfield.Fields) == 0 {
+					continue
+				}
+
 				prefix := ""
 				switch subfield.Name {
 				case "operation-metric":
+					if len(subfield.Fields[0].Fields) < 1 {
+						continue
+					}
 					prefix = subfield.Fields[0].Fields[0].GetStringValue()
 				case "class-stats":
+					if len(subfield.Fields[0].Fields) < 2 {
+						continue
+					}
```

Comment:
Same here. Wouldn't it be better to just leave the prefix empty? I don't know enough about this plugin to assess if it might be sensible to do not have subfields...

Reply:
heh I don't know enough about the plugin either :)

How about instead of the continue, we can return the default empty prefix?

Reply:
Yeah, but we need to also check in downstream functions as I think they will just unconditionally access the `Fields`...

Reply:
Looking at `parseContentField` it seems there are similar checks for the length already present.